### ISSUE TITLE
I need to get rid of the error.

### ISSUE THREAD

**Issue by danny69boc** [ZFTurbo/Music-Source-Separation-Training#16]
I am still a beginner in programming.
But I would love to use this wonderful program of yours.
However, after downloading it, I have all the necessary files and when I run the code, I get an error.
I don't know how to solve it for me.
Could you please tell me what is wrong?
This programming of yours is going to be a big dream come true for me.
Thanks.

(base) C:\Users\User\Desktop\Music-Source-Separation-Training-main>python train.py
Traceback (most recent call last):
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\train.py", line 515, in <module>
    train_model(None)
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\train.py", line 332, in train_model
    model, config = get_model_from_config(args.model_type, args.config_path)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\utils.py", line 14, in get_model_from_config
    with open(config_path) as f:
         ^^^^^^^^^^^^^^^^^
TypeError: expected str, bytes or os.PathLike object, not NoneType

**Comment by ZFTurbo** [comment]
You need to run program with parameters. For example:

```
python train.py \ 
    --model_type mel_band_roformer \ 
    --config_path configs/config_mel_band_roformer_vocals.yaml \
    --start_check_point results/model.ckpt \
    --results_path results/ \
    --data_path 'datasets/dataset1' 'datasets/dataset2' \
    --valid_path datasets/musdb18hq/test \
    --num_workers 4 \
    --device_ids 0
```

**Comment by danny69boc** [comment]
Thanks.
My command was wrong.
Perhaps I could have proceeded to the next step.
Thanks.
I typed the command as follows.
I'm really glad we've taken this one step further, thank you...

python train.py --model_type mdx23c --config_path configs/config_musdb18_mdx23c.yaml --start_check_point results/model_mdx23c_ep_168_sdr_7.0207.ckpt --results_path results/ --data_path 'datasets/Song 1' 'datasets/Song 2

The next error looked like this.
Is there a solution?

(base) C:\Users\User\Desktop\Music-Source-Separation-Training-main>python train.py --model_type mdx23c --config_path configs/config_musdb18_mdx23c.yaml --start_check_point results/model_mdx23c_ep_168_sdr_7.0207.ckpt --results_path results/ --data_path 'datasets/Song 1' 'datasets/Song 2
Instruments: ['vocals', 'bass', 'drums', 'other']
Use augmentation for training
Loading songs data from cache: results/metadata_1.pkl. If you updated dataset remove metadata_1.pkl before training!
Found tracks in dataset: 0
Start from checkpoint: results/model_mdx23c_ep_168_sdr_7.0207.ckpt
Traceback (most recent call last):
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\train.py", line 515, in <module>
    train_model(None)
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\train.py", line 366, in train_model
    load_not_compatible_weights(model, args.start_check_point, verbose=False)
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\train.py", line 56, in load_not_compatible_weights
    old_model = torch.load(weights)
                ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\serialization.py", line 1026, in load
    return _load(opened_zipfile,
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\serialization.py", line 1438, in _load
    result = unpickler.load()
             ^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\serialization.py", line 1408, in persistent_load
    typed_storage = load_tensor(dtype, nbytes, key, _maybe_decode_ascii(location))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\serialization.py", line 1382, in load_tensor
    wrap_storage=restore_location(storage, location),
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\serialization.py", line 391, in default_restore_location
    result = fn(storage, location)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\serialization.py", line 266, in _cuda_deserialize
    device = validate_cuda_device(location)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\serialization.py", line 250, in validate_cuda_device
    raise RuntimeError('Attempting to deserialize object on a CUDA '
RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.

**Comment by danny69boc** [comment]
Sorry, CUDA was able to solve the problem.
Let me ask you another question.

**Comment by danny69boc** [comment]
We are now at the stage where we can implement your program.
However, there is what appears to be the final error. I would like to know how to solve this.
Please let me know.

(base) C:\Users\User\Desktop\Music-Source-Separation-Training-main>python train.py --model_type mdx23c --config_path configs/config_musdb18_mdx23c.yaml --start_check_point results/model_mdx23c_ep_168_sdr_7.0207.ckpt --results_path results/ --data_path 'datasets/dataset1' 'datasets/dataset2
Instruments: ['vocals', 'bass', 'drums', 'other']
Use augmentation for training
Loading songs data from cache: results/metadata_1.pkl. If you updated dataset remove metadata_1.pkl before training!
Found tracks in dataset: 0
Start from checkpoint: results/model_mdx23c_ep_168_sdr_7.0207.ckpt
Use single GPU: [0]
Patience: 2 Reduce factor: 0.95 Batch size: 6 Grad accum steps: 1 Effective batch size: 6
Train for: 1000
Train epoch: 0 Learning rate: 9e-05
  0%|                                                                                         | 0/1000 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\train.py", line 515, in <module>
    train_model(None)
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\train.py", line 437, in train_model
    for i, (batch, mixes) in enumerate(pbar):
  File "C:\Users\User\anaconda3\Lib\site-packages\tqdm\std.py", line 1178, in __iter__
    for obj in iterable:
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\utils\data\dataloader.py", line 631, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\utils\data\dataloader.py", line 675, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\utils\data\_utils\fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\torch\utils\data\_utils\fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\dataset.py", line 488, in __getitem__
    res = self.load_random_mix()
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\dataset.py", line 163, in load_random_mix
    s1 = self.load_source(self.metadata, instr)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\Desktop\Music-Source-Separation-Training-main\dataset.py", line 149, in load_source
    track_path, track_length = random.choice(metadata)
                               ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\random.py", line 373, in choice
    raise IndexError('Cannot choose from an empty sequence')
IndexError: Cannot choose from an empty sequence

**Comment by ZFTurbo** [comment]
Try to remove "metadata_1.pkl". If it will not help check that your dataset paths are correct.

Log says:
`Found tracks in dataset: 0`

It means no tracks were found.

**Comment by danny69boc** [comment]
After deleting the specified metadata_1.pkl, I checked the file path and ran it again, but the same error appears.

The contents of dataset1 are the following 5 files.
bass.wav,drums.wav,mixture.wav,other.wav,vocals.wav
The song will be 5 minutes long.

Is there a problem with the sound source of the prepared pattern.... .1411kbps,44.1mbps standard WAV file!

**Comment by ZFTurbo** [comment]
Look how dataset must be structured:
https://github.com/ZFTurbo/Music-Source-Separation-Training/blob/main/docs/dataset_types.md

It must be several folders inside. Each folder contains "bass.wav,drums.wav,mixture.wav,other.wav,vocals.wav"

**Comment by danny69boc** [comment]
Currently stored in Type 1 format.
-datasets
--- Song 1:.
------ vocals.wav  
------ bass.wav 
------ drums.wav
------ other.wav
------ mixture.wav

Is there a problem with this structure?
Do I need to create all folders from Type 1 to Type 4?

**Comment by ZFTurbo** [comment]
Print the path to vocals.wav in your system.

**Comment by danny69boc** [comment]
"C:\Users\User\Desktop\Music-Source-Separation-Training-main\datasets\Song 1\vocals.wav"

Is it wrong to specify this location as follows?

--data_path 'datasets/Song 1/' 'datasets/Song 2/'

**Comment by ZFTurbo** [comment]
It must be:
`--data_path 'datasets/'`

**Comment by danny69boc** [comment]
I entered the following command.
It should not be wrong as a string.What could possibly go wrong...

python train.py --model_type mdx23c --config_path configs/config_musdb18_mdx23c.yaml --start_check_point results/model_mdx23c_ep_168_sdr_7.0207.ckpt --results_path results/ --data_path 'datasets/Song1/' 'datasets/Song2/'

**Comment by ZFTurbo** [comment]
Try like this:

```
python train.py --model_type mdx23c --config_path configs/config_musdb18_mdx23c.yaml --start_check_point results/model_mdx23c_ep_168_sdr_7.0207.ckpt --results_path results/ --data_path 'datasets/' 
```

**Comment by danny69boc** [comment]
The result is the same, it doesn't work...

Is there something wrong with creating the dataset?

**Comment by danny69boc** [comment]
Set up from scratch only once more, entered your proceed code and tried.
Then I checked the string, made sure nothing was wrong, and entered it.
The string you taught me was the same, and the last thing I tried was to see if I could read the first data in the data set in the first place!

But the result is still the same!
Just to be sure, I tried to replace the first data in the data set.
However, it ended up being blank and would not read, and I keep getting the same error.
I have no idea how to solve this.
I have tried everything I can do.
